mantidproject/mantid
# ConvertUnits could corrupt inplace workspaces on error

Several validation checks were taking place after the output workspace had already been altered.  for In place workspaces this meant the workspace was altered even though the algorithm failed.

The changes delay the creation of the output workspace and changes until all of the validation has passed.

**To test:**
1. All unit tests and system tests must pass
2. Create or load a workspace
3. Try to convert to DeltaE, or SpinEchoLength with an Elastic Emode and not setting eFixed.
4. Check that the input workspace has not changed (check the units of the ws).

Fixes #16794
### RELEASE NOTES

Either edit the file in docs/source/release/ui.rst

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
